### PR TITLE
File loading improvements

### PR DESCRIPTION
**Pull request by haphzd**

### CODE REVIEW

**Comment by JayFoxRox** [comment]
Nice work :)

There are various other places in the code where binary files (script for example) are opened without that flag, could you attempt to change all of them in your PR?

We should probably discuss wether we also want to open "*.dat" files as binary too because the stupid newline stuff windows does is just cancerous behaviour..
We'd never be guaranteed that our code is portable.

**Comment by danhedron** [comment]
Is there any progress on this?

**Comment by haphzd** [comment]
Yes, don't merge yet.

File: rwgame/RWGame.cpp [hunk 1]
```diff
@@ -111,7 +111,6 @@ RWGame::RWGame(int argc, char* argv[])
 	data->loadIMG("/models/gta3");
 	//engine->data.loadIMG("/models/txd");
 	data->loadIMG("/anim/cuts");
-	data->loadTXD("/models/hud.txd");
```
Comment (JayFoxRox) [inline rwgame/RWGame.cpp]: might be a stupid question, but where is it loaded now?

Comment (danhedron) [inline rwgame/RWGame.cpp]: `GameData::load()` should load that file

**Comment by JayFoxRox** [comment]
I don't really like that file abstraction we currently have (because we only seem to have one instance to look in every folder which (luckily!) works for GTA3). We'll need an issue for that soon.
Your current changes are fine though.

@haphzd: Can you still grep for the ifstream stuff and add `, std::ios::binary` everywhere so we can get this merged?

**Comment by haphzd** [comment]
Enough for now, I think.

**Comment by danhedron** [comment]
Looks good to me, tested locally and it works fine.

File: tests/test_SaveGame.cpp [hunk 2]
```diff
@@ -55,7 +55,7 @@ BOOST_AUTO_TEST_CASE(test_load_game)
 	GameState state;
 	GameWorld world();
 	SCMOpcodes s;
-	auto file = Global::get().d->loadSCM("data/main.scm");
+	auto file = Global::get().d->loadSCM("main.scm");
```
Comment (JayFoxRox) [inline tests/test_SaveGame.cpp]: Why is this part of this commit?

File: rwlib/source/platform/FileIndex.cpp [hunk 3]
```diff
@@ -155,7 +155,7 @@ FileHandle FileIndex::openFile(const std::string& filename)
 	}
 	else
 	{
-		std::ifstream dfile(fsName.c_str());
+		std::ifstream dfile(fsName.c_str(), std::ios::binary);
```
Comment (JayFoxRox) [inline rwlib/source/platform/FileIndex.cpp]: Please merge this with the other ifstream changes.

**Comment by JayFoxRox** [comment]
Tested fine locally. I just think the commit structure has to improve a little.
Especially the SCM change breaks tests for one commit when it's really easy to avoid it.

So if you could fix what I mentioned this LGTM. Thanks!

**Comment by danhedron** [comment]
Thanks for fixing this. Looks good to me now.

File: rwengine/src/engine/GameData.cpp [hunk 4]
```diff
@@ -341,22 +341,12 @@ void GameData::loadHandling(const std::string& path)
...
-
-	if(! f.is_open() ) return nullptr;
-
-	f.seekg(0, std::ios_base::end);
-	unsigned int size = f.tellg();
-	f.seekg(0);
-
-	char* buff = new char[size];
-	f.read(buff, size);
+	auto scm_h = openFile(name);
```
Comment (JayFoxRox) [inline rwengine/src/engine/GameData.cpp]: nit: Dan likes to camelCase variable names.

(as this is the only thing I can nit about this is fine to me, we'll probably catch it later when loading SCM from the proper path with a better IO abstraction)

**Comment by JayFoxRox** [comment]
- [x] Game still works
- [x] Doesn't contain malware
- [x] Follows our code style

LGTM unless you still want to change anything @haphzd  :)

**Comment by danhedron** [comment]
Thanks. I appreciate the work in getting things working on Windows.

**Pull request merged**